src/krill/systems.rs

```diff
@@ -1,84 +1,305 @@
+        commands.spawn(KrillBundle {
+            krill: Krill,
+            name: Name::new(KRILL),
+            sprite: SpriteSheetBundle {
+                sprite: TextureAtlasSprite {
+                    index: 0,
+                    custom_size: Some(Vec2::new(KRILL_RADIUS * 2., KRILL_RADIUS * 2.)),
                     ..Default::default()
                 },
+                texture_atlas: image_assets.krill.clone(),
+                transform: Transform::from_translation(Vec3::new(random_x, random_y, 1.)),
+                ..Default::default()
+            },
+            animation_indices: AnimationIndices { first: 0, last: 1 },
+            animation_timer: AnimationTimer(Timer::from_seconds(0.1, TimerMode::Repeating)),
+            collider: Collider::ball(KRILL_RADIUS),
+            collision_group: CollisionGroups {
+                memberships: KRILL_COLLISION_GROUP,
+                filters: Group::complement(KRILL_COLLISION_GROUP),
+            },
+            ridgid_body: KRILL_RIGID_BODY,
+            restitution: Restitution {
+                coefficient: KRILL_RESTITUTION_COE,
+                combine_rule: CoefficientCombineRule::Max,
+            },
+            friction: Friction {
+                coefficient: KRILL_FRICTION_COE,
+                combine_rule: CoefficientCombineRule::Min,
             },
-            Name::new("Krill"),
-            AnimationIndices { first: 0, last: 1 },
-            AnimationTimer(Timer::from_seconds(0.1, TimerMode::Repeating)),
-        ));
+            boid: BoidBundle {
+                boid: Boid,
+                acceleration: Acceleration { vec: Vec2::ZERO },
+                velocity: Velocity::linear(random_starting_vel * random_starting_speed),
+                align: Align { vec: Vec2::ZERO },
+                seperation: Seperation { vec: Vec2::ZERO },
+                cohesion: Cohesion { vec: Vec2::ZERO },
+            },
+        });
     }
 }
 
 pub fn debug_krill(
     mut debug_event_reader: EventReader<DebugEvent>,
-    krill_query: Query<&Transform, With<Krill>>,
+    krill_query: Query<(&Transform, &Velocity, &Acceleration), With<Krill>>,
 ) {
     for _event in debug_event_reader.read() {
-        info!("{:?}", krill_query);
-        let krill_transform = krill_query.get_single().unwrap();
-        info!("{:?}", krill_transform);
-        info!("{:?}", krill_transform.forward());
+        for (krill_transform, krill_vel, krill_acc) in krill_query.iter() {
+            info!("{:?}", krill_transform);
+            info!("{:?}", krill_vel);
+            info!("{:?}", krill_acc);
+        }
     }
 }
 
 pub fn krill_idle_movement(mut krill_query: Query<&mut Transform, With<Krill>>, time: Res<Time>) {
     for mut krill_transform in krill_query.iter_mut() {
         const IDLE_HIEGHT_SCALAR: f32 = 0.005;
         const IDLE_FREQ_SCALAR: f32 = 0.1;
+        const IDLE_PERIOD_SCALAR: f32 = 20.;
```

Comment:
I would move these to outside the function if you want them to be constants. It's fine if you want to put it right above the function, but I'm not sure what the behavior of having this inside the loop would be if any.